src/native/http_connection_manager.c

```diff
@@ -332,7 +341,8 @@ static void s_destroy_connection_binding(struct aws_http_connection_binding *bin
         return;
     }
 
-    JNIEnv *env = aws_jni_get_thread_env(binding->jvm);
+    /********** JNI ENV ACQUIRE **********/
+    JNIEnv *env = aws_jni_acquire_thread_env(binding->jvm);
```

Comment:
`s_destroy_connection_binding()` is a helper function, not a callback, we could just pass `env` in as an argument